### ISSUE TITLE
Launchd Missing Hidden Launch Agents/Daemons

### ISSUE THREAD

**Issue by calhall** [osquery/osquery#7703]
# Blueprint

**TL;DR**: the current implementation of launchd does not identify LaunchAgents/Daemons that originate from hidden files (i.e. `/Library/LaunchAgents/.hiddenfile.test.plist`.

As demonstrated within [this article](https://www.welivesecurity.com/2022/07/19/i-see-what-you-did-there-look-cloudmensis-macos-spyware/), malware has been observed to use hidden plist files to create less visible LaunchAgents/Daemons. This is a very simple evasion technique, however the current implementation of osquery does not detect these LaunchAgents/Daemons as shown below:
![image](https://user-images.githubusercontent.com/27683329/180995024-6934b924-eb00-4bd1-832d-ad22fb5afbc1.png)

As macOS allows LaunchAgents/Daemons to be loaded from hidden `.plist` files, this creates a visibility gap within osquery's launchd table.

The two potential solutions I imagine would be:
* Implement the launchd table generation to utilise macOS APIs to gather information about loaded LaunchAgents/Daemons, rather than simply monitoring specific directories (high effort)
* Add the capability to list hidden files within the monitored LaunchAgents/Daemons directories (low effort)

Given that the latter addresses this specific issue,and is feasible to implement with the current osquery setup, I am going to push a PR to introduce this capability.
